index.js

```diff
@@ -185,3 +197,72 @@ function createRedirectDirectoryListener() {
     res.end(msg)
   }
 }
+
+/**
+ * Modifies the response header for gzipped assets
+ * @private
+ */
+
+function setGzipHeaders(res, path) {
+  var type = mime.lookup(path)
+  var charset = mime.charsets.lookup(type)
+
+  res.setHeader('Content-Type', type + (charset ? '; charset=' + charset : ''))
+  res.setHeader('Content-Encoding', 'gzip')
+  res.setHeader('Vary', 'Accept-Encoding')
+}
+
+/**
+ * Searches root folder on init for gzipped assets
+ * @private
+ */
+
+function createCache(root) {
+  var cache = Object.create(null)
+
+  find.fileSync(/\.gz$/, root).forEach(function(file) {
```

Comment:
A feature of this module is that the contents of your static directory can change without a server restart. Can a better solution be applied for this that will take that into account?